src/workerd/tests/bench-builtin-modules.c++

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include <workerd/tests/bench-tools.h>
+#include <workerd/tests/test-fixture.h>
+
+// A benchmark for loading built-in modules.
+
+namespace workerd {
+
+static CompatibilityFlags::Builder createCompatibilityFlags() {
+  static capnp::MallocMessageBuilder flagsBuilder;
+  static auto featureFlags = flagsBuilder.initRoot<CompatibilityFlags>();
+  featureFlags.setNodeJsCompat(true);
+  featureFlags.setNodeJsCompatV2(true);
+  return featureFlags;
+}
+
+class BuiltinModulesBenchmark: public benchmark::Fixture {
+public:
+  virtual ~BuiltinModulesBenchmark() noexcept(true) {}
+
+  void SetUp(benchmark::State& state) noexcept(true) override {
+    static auto flags = createCompatibilityFlags();
+    TestFixture::SetupParams params = {.featureFlags = flags.asReader(), .mainModuleSource = R"(
+        import assert from "node:assert";
+        export default {
+          async fetch(request) {
+            let buffer;
+            for (let i = 0; i < 100; i++) {
+              buffer = await import("node:buffer");
+            }
+            assert.ok(buffer);
+            return new Response("OK");
+          },
+        };
+      )"_kj};
+    fixture = kj::heap<TestFixture>(kj::mv(params));
+  }
```

Comment:
My concern here is that if this is meant to test things like the efficacy of compile cache, it's not clear this will actually be a useful benchmark. The internal module will end up being compiled and cached only once per worker isolate, with the compile cache being populated and reused only when another isolate is loaded that imports the same module. We wouldn't see any performance impact on the first isolate and would only see it on the subsequent additional isolates as reduced cold start times.